citar-file.el

```diff
@@ -128,14 +128,23 @@ Example: ':/path/to/test.pdf:PDF'."
          (mapcar (apply-partially #'expand-file-name fn) dirs)))
      parts)))
 
-(defun citar-file--parse-file-field (entry fieldname &optional dirs)
+(defun citar-file--parse-file-field (entry fieldname &optional dirs extensions)
   "Return files listed in FIELDNAME of ENTRY.
-File names are expanded relative to the elements of DIRS."
+File names are expanded relative to the elements of DIRS.
+
+Filter by EXTENSIONS when present."
   (unless dirs (setq dirs (list "/")))  ; make sure DIRS is non-nil
-  (when-let ((filefield (citar-get-value fieldname entry)))
-    (seq-mapcat (lambda (parser)
-                  (funcall parser dirs filefield))
-                citar-file-parser-functions)))
+  (let* ((filefield (citar-get-value fieldname entry))
+         (files
+          (delete-dups
```

Comment:
I'm not clear why I needed to add this.

Reply:
Probably because each of the file parsers returns the same list of files, you'll end up with two copies of everything. Deleting the duplicates seems like a reasonable way of handling that.